packages/babel-plugin-transform-object-rest-spread/src/index.js

```diff
@@ -65,9 +65,13 @@ export default function({ types: t }) {
 
   //expects path to an object pattern
   function createObjectSpread(path, file, objRef) {
-    const last = path.get("properties").pop(); // note: popping does not mean removal from path
-    const restElement = t.clone(last.node);
-    last.remove(); // remove restElement
+    const props = path.get("properties");
+    const last = props[props.length - 1];
+    const restElement = last.node.type === "RestElement" && t.clone(last.node);
```

Comment:
I don't understand how this can ever be false.

Reply:
This code wasn't always popping off a `RestElement` before. You can add `if (! restElement) debugger` and run the test with
```sh
TEST_DEBUG=true TEST_ONLY=babel-plugin-transform-object-rest-spread make test-only
```
to convince yourself that it can happen, though I agree it's worth understanding why.

I think this happens when [this `RestElement` visitor](https://github.com/babel/babel/blob/4e7961dcf8104e390b39b534a3ec39b795a7a0c1/packages/babel-plugin-transform-object-rest-spread/src/index.js#L146) encounters the `RestElement` nested inside the `ArrayPattern` (that is, `[b, ...arrayRest]`), but then finds the nearest `ObjectPattern` parent [here](https://github.com/babel/babel/blob/4e7961dcf8104e390b39b534a3ec39b795a7a0c1/packages/babel-plugin-transform-object-rest-spread/src/index.js#L191).

Reply:
That looks like it's it. Instead, we should probably be checking if the `RestElement` is direct child of a `ObjectPattern` and early returning if not.

Reply:
Let me see if that fixes it, and I'll push a follow-up commit.